pkg/quantile/agent.go

```diff
@@ -57,16 +61,32 @@ func (a *Agent) Insert(v float64) {
 	a.flush()
 }
 
-// InsertN inserts v, n times into the sketch.
-func (a *Agent) InsertN(v float64, n uint) {
-	a.Sketch.Basic.InsertN(v, n)
-
-	for i := 0; i < int(n); i++ {
-		a.Buf = append(a.Buf, agentConfig.key(v))
+// InsertInterpolate linearly interpolates counts from lower (l) to upper (u)
+func (a *Agent) InsertInterpolate(l float64, u float64, n uint) {
```

Comment:
nit: might be worth unit testing this even though it's covered by check_sampler tests